docs/trace/extending-the-sdk/MyResourceDetector.cs

```diff
@@ -0,0 +1,55 @@
+// <copyright file="MyResourceDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.Resources;
+
+internal class MyResourceDetector : IResourceDetector
+{
+    public const string EnVarkey = "myEnVarkey";
+
+    public Resource Detect()
+    {
+        var resource = Resource.Empty;
+        if (this.LoadString(EnVarkey, out string envAttributeVal))
```

Comment:
keep this as simple as possible. I'd recommend to simply return a new List of attributes. Don't need to show env variable parsing or anything here.

Reply:
I think it serves as a good example for the users to provide a more wholistic usage on where to get the resource attributes, at lease for me I think it is good to have information. But I'm open to other suggestions.

Reply:
eg: Sampler (doc has the simplest one, but links to other official ones from this repo)